File: 프로그래머스/2/17687. ［3차］ n진수 게임/［3차］ n진수 게임.java
class Solution {
    public String change(int n, int q) {
        StringBuilder result = new StringBuilder();
        if (n == 0)
            return "0";
        while (n != 0) {
            if (n % q >= 10)
                result.append((char) ('A' + (n%q) - 10));
            
            else
                result.append(n % q);
            n /= q;
        }
        return result.reverse().toString();
    }
    
    public String solution(int n, int t, int m, int p) {
        String answer = "";
        int count = 0;  // n진수로 바꾸어야 할 수
        int total = 1;  //n진수로 바꾼 길이의 총 합
        while (answer.length() <= t) {
            // String strN = change(count, n);
            String strN = Integer.toString(count, n);
            for (int i=0; i<strN.length(); i++) {
                if ((total+i) % m == p % m) // m=p일 때를 위해
                    answer += strN.charAt(i);
            }
            count++;
            total += strN.length();
        }
        return answer.toUpperCase().substring(0, t);
    }
}

File: 프로그래머스/2/17687. ［3차］ n진수 게임/［3차］ n진수 게임.py
def change(n, q):
    result = ''
    if n == 0:
        return '0'
    while (n != 0):
        if (n%q) >= 10:
            result += chr(ord('A') + (n%q) -10)
        else:
            result += str(n % q)
        n //= q
    return result[::-1]

def solution(n, t, m, p):
    answer = ''
    count = 0 # n진수로 바꾸어야 할 수
    total = 1 # n진수로 바꾼 길이의 총 합
    while len(answer) <= t:
        strN = change(count, n)
        for idx, ch in enumerate(strN):
            if (total+idx) % m == p % m: # m=p일 때를 위해
                answer += ch
        total += len(strN)
        count += 1
    return answer[:t]